sympy/printing/repr.py

```diff
@@ -140,7 +140,13 @@ def _print_Sum2(self, expr):
                                            self._print(expr.a), self._print(expr.b))
 
     def _print_Symbol(self, expr):
-        return "%s(%s)" % (expr.__class__.__name__, self._print(expr.name))
+        d = expr._assumptions.generator
```

Comment:
@moorepants I access a private `_assumptions` of expr here.  Is this ok?  Or do I need to write some sort of getter for `Symbol`?  If so, what should it be called?  `expr.assumptions_generator`?  `expr.user_assumptions`?  (I don't really understand why the current getter is called `assumptions0`)